### PR TITLE
feat: Enable URL drop functionality in Image Manager

### PR DESCRIPTION
**Pull request by istarkov**
## Description

https://github.com/webstudio-is/webstudio/assets/5077042/ea38835c-ec5c-4333-a982-f5bb9580d9e0

URLs can be dropped on Image Manager

+ small fixes.
- Files without extensions broke FileName control
- svg files were not shown on local builder after upload

## Steps for reproduction

Open Image Manager, drag urls

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env` file

### CODE REVIEW

File: apps/builder/app/builder/shared/assets/use-assets.tsx [hunk 1]
```diff
@@ -31,68 +33,108 @@ const setAsset = (asset: Asset) => {
...
+    | {
+        source: "file";
+        file: File;
+      }
+    | {
+        source: "url";
+        url: URL;
+      }
+  )
+>;
```
Comment (TrySound) [inline apps/builder/app/builder/shared/assets/use-assets.tsx:52]: not simple 😉

Comment (istarkov) [reply apps/builder/app/builder/shared/assets/use-assets.tsx:52]: After Simplify on hover it's simple type ;-)

image

File: apps/builder/app/builder/shared/assets/image-formats.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+const extensionToMime = new Map([
+  [".gif", "image/gif"],
+  [".ico", "image/x-icon"],
+  [".jpeg", "image/jpeg"],
+  [".jpg", "image/jpeg"],
+  [".png", "image/png"],
+  [".svg", "image/svg+xml"],
+  [".webp", "image/webp"],
+] as const);
```
Comment (TrySound) [inline apps/builder/app/builder/shared/assets/image-formats.ts:11]: looks like we have something similar in asset manager package

File: apps/builder/app/routes/rest.assets.$name.tsx [hunk 3]
```diff
@@ -15,11 +20,37 @@ export const action = async (
...
-      const asset = await uploadFile(
-        params.name,
-        request.body,
-        createAssetClient()
-      );
+      let body = request.body;
+
+      const contentType = request.headers.get("Content-Type");
+
+      if (contentType?.includes("application/json")) {
```
Comment (TrySound) [inline apps/builder/app/routes/rest.assets.$name.tsx:27]: why includes and not === ?

File: apps/builder/app/builder/shared/assets/image-formats.ts (hunk 2 above)
Comment (istarkov) [reply apps/builder/app/builder/shared/assets/image-formats.ts:11]: Yep but there ton of types we don't need in UI is supported

**Review by TrySound — APPROVED** [review]

File: apps/builder/app/routes/rest.assets.$name.tsx (hunk 3 above)
Comment (istarkov) [reply apps/builder/app/routes/rest.assets.$name.tsx:27]: `Content-Type: application/json; charset=UTF-8`

**Review by kof — APPROVED** [review]
Works well. One thing that could be even better is drop target detection without opening the image manger and then opening image manager automatically as soon as anything is being dragged onto builder

**Pull request merged**